dem.cc

```diff
@@ -90,24 +108,90 @@ void update_acceleration(int jm, int js) {
     force[0] = force_normal * normal[0] + force_tangent * tangent[0];
     force[1] = force_normal * normal[1] + force_tangent * tangent[1];
     // updatig accerelations
-    grains[jm].ax = force[0] / grains[jm].m;
-    grains[jm].ay = force[1] / grains[jm].m;
-    grains[jm].alpha =
+    grains[jm].ax += force[0] / grains[jm].m;
+    grains[jm].ay += force[1] / grains[jm].m;
+    grains[jm].alpha +=
         (-force_tangent * grains[jm].r + rolling_moment) / grains[jm].inertia;
-    // std::cout<<delta<<"\t"<<force_normal<<"\t"<<force[1]<<"\t"<<grains[jm].ay<<std::endl;
-    // adding the assigned gravity to the accelerations
-    grains[jm].ax += grains[jm].gx;
-    grains[jm].ay += grains[jm].gy;
   }
 }
 
+// function to calculate the forces between particle jm and wall w and update
+// the acceleration of jm in 2 directions.
+void update_acceleration_grain_wall(int jm, int w) {
+
+  const double distance =
+      (walls[w].tan_slope * grains[jm].xc - grains[jm].yc -
+       walls[w].tan_slope * walls[w].point[0] + walls[w].point[1]) /
+      sqrt(walls[w].tan_slope * walls[w].tan_slope + 1.);
+  double delta;
+  if (walls[w].tangent[0] > 0.) {
```

Comment:
What happens if it is exactly 0?

Reply:
I'll fix that

Reply:
With the current implementation it will never be 0. Even if you set the slope_angle to 90, cos(90) will be a very small number (6e-17), and the code will still work and result the correct response. The sign of cos(90) will determine whether the wall normal is to the right or to the left. However, if the user uses {1,0} for the normal vector rather than {sin(90),cos(90)} then the code won't work since tan_slope will be NaN and distance will be NaN. So in that case I think I will need an if statement when defining tan_slope, but I can't have an if statement inside a struct, so I don't know what to do about that.

Reply:
OK that's fine then